src/main/scala/edu/ie3/simona/agent/participant/evcs/EvcsAgentFundamentals.scala

```diff
@@ -466,6 +468,51 @@ protected trait EvcsAgentFundamentals
     )
   }
 
+  /** Filters schedules for a given Ev and returns all entries within range of a
+    * given Tick
+    *
+    * @param ev
+    *   the EvModelWrapper for which the the schedules should be filtered
+    * @param lastTick
+    *   the last Tick
+    * @param currentTick
+    *   the current Tick
+    * @param schedules
+    *   the map of schedules that should be filtered
+    * @return
+    *   If Ev is in schedules:
+    *   - the [[ChargingSchedule]] of that Ev for the currentTick or if there
+    *     isn't
+    *   - an default [[ChargingSchedule]] with zero Power will be returned If
+    *     the Ev isn't in schedules a [[RuntimeException]] will be thrown since
+    *     it should be in the schedules.
+    */
+
+  private def filterSchedulesForTick(
+      ev: EvModelWrapper,
+      lastTick: Long,
+      currentTick: Long,
+      schedules: EvcsModel.ScheduleMap,
+  ): EvcsModel.ScheduleMap = {
+
+    // in order to create 0kW entries for EVs that do not
+    // start charging right away at lastTick, create mock
+    // schedule entries that end before lastTick
+
+    val defaultEntry = ScheduleEntry(lastTick, lastTick, Kilowatts(0d))
+
+    val schedule = schedules.getOrElse(ev.uuid, SortedSet(defaultEntry))
+
+    val updatedSchedule = schedule.map { entry =>
+      if (currentTick >= entry.tickStart && currentTick <= entry.tickStop)
```

Comment:
Why do we filter out schedule entries that start after the current tick? They'll still be valid at some point in the future, right?

Reply:
Yes, I agree. But wouldn't they reappear in the next tick? I don't remember the details, maybe this change overlapped with some other change, so I got the impression that this was necessary. However, tests show that this change is not necessary at all, so I will rollback here.